src/scanner.c

```diff
@@ -1,18 +1,46 @@
+#include <string.h>
 #include <tree_sitter/parser.h>
 #include <wctype.h>
 
 enum TokenType {
   STRING_CONTENT,
-  RAW_STRING_LITERAL,
+  RAW_STRING_LITERAL_OPEN,
+  RAW_STRING_LITERAL_CONTENT,
+  RAW_STRING_LITERAL_CLOSE,
   FLOAT_LITERAL,
   BLOCK_COMMENT,
 };
 
-void *tree_sitter_rust_external_scanner_create() { return NULL; }
-void tree_sitter_rust_external_scanner_destroy(void *p) {}
+struct ScannerState {
+  unsigned opening_hash_count;
+};
```

Comment:
A struct for one field is a bit overkill, let's just allocate a uint32_t instead (I'd rather not use unsigned)

Reply:
Are you sure about this? Personally I feel keeping it in a struct makes it explicitly clear that this is the only state we keep track of.

Reply:
alright yeah, if we do need to add more fields later on for whatever reason it'd be easier. Can we declare it as a typedef, use uint32_t, and call it just Scanner though?
```c
typedef struct {
    uint32_t opening_hash_count;
} Scanner
```

Reply:
I am not a big fan of anonymous structs, would f6caf4ae21733980406e8982c4349ea44181731f work as a compromise?

If not I am okay with removing the name. It's mostly for the sake of things like lsp hover information.